ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh

```diff
@@ -51,17 +51,51 @@ function cli_Login() {
 }
 
 function gatherLBs() {
-    local lbs aps
+    local lbs aps name state ip_id public_ip_address private_ip_address public_ip_name temp_file backend_ids
+    echo ""
     run_command "az network lb list --resource-group $RESOURCE_GROUP > $OUTPUT_DIR/lbs.json"
-    run_command "az network lb list --resource-group $RESOURCE_GROUP -o tsv"
+    run_command "az network lb list --resource-group $RESOURCE_GROUP -o table"
     lbs="$(az network lb list -g $RESOURCE_GROUP | jq -r '.[].name')"
     for lb in $lbs; do
-        echo "loadbalance: $lb"
+        echo -e "\n=> loadbalance: $lb"
+
+        run_command "az network lb frontend-ip list --lb-name $lb -g $RESOURCE_GROUP > $OUTPUT_DIR/frontend-ips.json"
+        temp_file=$(mktemp)
+        echo -e "Name\tProvisioning State\tPublic IP Name\tPublic IP Address\tPrivate IP Address" >> "$temp_file"
+        echo -e "----\t------------------\t--------------\t----------------\t-----------------" >> "$temp_file"
+        while IFS=$'\t' read -r name state public_ip_id private_ip; do
+            public_ip_name="N/A"
+            public_ip_address="N/A"
+            private_ip_address="N/A"
+            if [ "$public_ip_id" != "null" ]; then
+                public_ip_name=$(basename "$public_ip_id")
+                public_ip_address=$(az network public-ip show --ids "$public_ip_id" --query ipAddress -o tsv 2>/dev/null)
+                [ -z "$public_ip_address" ] && public_ip_address="Not Found"
```

Comment:
when `$public_ip_address` is empty, I guess the script will exit here, it's better to use if statement instead.

Reply:
updated.